theano/tensor/nnet/abstract_conv.py

```diff
@@ -1663,18 +1716,27 @@ def conv(self, img, kern, mode="valid", dilation=1, num_groups=1):
             raise ValueError(
                 'invalid dilation {}, expected {} values'.format(dilation,
                                                                  self.convdim))
+        if unshared and direction == "backprop weights":
+            if mode != "valid":
+                raise ValueError('conv mode for unshared backprop wrt weights must be "valid"')
+            # Do a transpose later to bring it to required shape
+            out_shape = (img.shape[0], kern.shape[0],
+                         kern.shape[2], kern.shape[3],
+                         img.shape[2] - kern.shape[2] + 1,
+                         img.shape[3] - kern.shape[3] + 1)
```

Comment:
@nouiz Yes, that's correct. `get_conv_output_shape()`, `get_conv_gradinputs_shape()` and `get_conv_gradweights_shape()` methods all support unshared convolution. Only `get_conv_gradweights_shape()` needs an additional `unshared` parameter for this, because the type of convolution is irrelevant for the other two. The shape of those is only influenced by the kernel width and height, but it's the same for shared and unshared convolutions.

The shape that's computed here is very much a special case, which is only relevant when it's combined with the other trickery below. It's not something that you would want to compute in practice, so I think computing it directly is fine.